src/main/java/com/mcmoddev/lib/util/MMDResourceLocation.java

```diff
@@ -0,0 +1,29 @@
+package com.mcmoddev.lib.util;
+
+import java.util.UUID;
+import net.minecraft.util.ResourceLocation;
+
+public class MMDResourceLocation extends ResourceLocation {
+    private final String entryNamespace;
+    private final String entryType;
+    private final String entryName;
+    
+    public MMDResourceLocation(String namespace, String entryType, String entryName ) {
+        super( "mmdlib", UUID.nameUUIDFromBytes(String.format("%s:%s/%s", namespace, entryType, entryName).getBytes()).toString() );

```

Comment:
Would advise on using `getBytes(Charsets.UTF_8)` to not rely on the default charset.